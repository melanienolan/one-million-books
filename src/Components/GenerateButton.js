import React from 'react';
import PropTypes from 'prop-types';

const GenerateButton = ({ number, generateBooks }) => {
  return (
    <button
      className={number < 1000000 ? 'button--generate-small' : 'button--generate-large'}
      onClick={() => generateBooks(number)}>
      {number}
    </button>
  );
};

export default GenerateButton;

GenerateButton.propTypes = {
  number: PropTypes.number,
  generateBooks: PropTypes.func
};
