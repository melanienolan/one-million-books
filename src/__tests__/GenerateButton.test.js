import React from 'react';
import GenerateButton from '../Components/GenerateButton';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('renders generate button component without crashing', () => {
  Enzyme.shallow(<GenerateButton number={10} />);
});
it('renders generate button component with 10', () => {
  const GenerateButtonComponent = Enzyme.shallow(<GenerateButton number={10} />);
  expect(toJson(GenerateButtonComponent)).toMatchSnapshot();
});
it('renders generate button component with 1000000', () => {
  const GenerateButtonComponent = Enzyme.shallow(<GenerateButton number={1000000} />);
  expect(toJson(GenerateButtonComponent)).toMatchSnapshot();
});
