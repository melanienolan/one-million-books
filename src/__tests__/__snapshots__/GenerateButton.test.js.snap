// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders generate button component with 10 1`] = `
<button
  className="button--generate-small"
  onClick={[Function]}
>
  10
</button>
`;

exports[`renders generate button component with 1000000 1`] = `
<button
  className="button--generate-large"
  onClick={[Function]}
>
  1000000
</button>
`;
